client/src/modules/depots/modals/depot.modal.js

```diff
@@ -37,4 +41,12 @@ function DepotModalController($state, Depots, ModalService, Notify) {
   function closeModal() {
     $state.go('depots');
   }
+
+  function clean(depot) {
```

Comment:
In the future, this would be better placed in the `DepotService`.  It doesn't really have anything to do with the view, and is only called when `DepotSerivce.create()` or `DepotService.update()` are called.

Reply:
You could also delete `hasLocation` here, and that would save you the work of putting `delete hasLocation` on the server.